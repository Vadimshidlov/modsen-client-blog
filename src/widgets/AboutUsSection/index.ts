export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
