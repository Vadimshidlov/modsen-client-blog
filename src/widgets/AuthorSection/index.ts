export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
