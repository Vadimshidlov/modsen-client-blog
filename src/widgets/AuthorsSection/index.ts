export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
