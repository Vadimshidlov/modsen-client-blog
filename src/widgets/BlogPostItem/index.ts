export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
