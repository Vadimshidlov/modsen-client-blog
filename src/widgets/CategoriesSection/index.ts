export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
