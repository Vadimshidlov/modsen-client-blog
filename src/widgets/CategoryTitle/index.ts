export { CategoryTitle } from '@/widgets/CategoryTitle/ui/CategoryTitle';
