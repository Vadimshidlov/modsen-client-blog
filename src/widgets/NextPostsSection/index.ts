export { NextPostsSection } from '@/widgets/NextPostsSection/ui/NextPostsSection';
