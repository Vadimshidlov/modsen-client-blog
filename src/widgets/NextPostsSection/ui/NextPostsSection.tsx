import React from 'react';
import { useTranslations } from 'next-intl';
import styles from '@/widgets/NextPostsSection/ui/NextPostsSection.module.scss';
import { POSTS } from '@/shared/constants/posts';
import { NextPostItem } from '@/widgets/NextPostsSection/ui/NextPostItem';
import { NextPostsSectionPropsType } from '@/shared/types';

export function NextPostsSection({ activePostId }: NextPostsSectionPropsType) {
  const t = useTranslations('NextPostsSection');
  const filteredPosts = POSTS.filter((postData) => postData.id !== activePostId);

  return (
    <section className={styles.nextPostSection}>
      <h2 className={styles.sectionTitle}>{t('title')}</h2>
      <div className={styles.nextPostsContainer}>
        {filteredPosts.slice(0, 3).map(({ id, text, postDate, title, author, img }) => (
          <NextPostItem
            postId={id}
            postText={text}
            postTitle={title}
            postDate={postDate}
            postAuthor={author}
            postImage={img}
            key={id}
          />
        ))}
      </div>
    </section>
  );
}
