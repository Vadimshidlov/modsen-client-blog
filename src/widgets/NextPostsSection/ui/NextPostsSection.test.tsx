import React from 'react';
import { render } from '@testing-library/react';
import { NextPostsSection } from '@/widgets/NextPostsSection';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('NavigationBar tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<NextPostsSection activePostId={1} />);
    expect(container).toMatchSnapshot();
  });
});
