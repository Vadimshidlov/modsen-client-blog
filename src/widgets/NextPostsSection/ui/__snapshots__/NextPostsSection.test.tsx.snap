// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar tests should matches snapshot 1`] = `
<div>
  <section
    class="nextPostSection"
  >
    <h2
      class="sectionTitle"
    >
      Mocked title
    </h2>
    <div
      class="nextPostsContainer"
    >
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="social-pic"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <span
          class="postDateInfo variable"
        >
          Mocked authorTag
           
          <span>
            Dianne Russell
          </span>
           
          Aug 23, 2024
        </span>
        <a
          href="/posts/2"
        >
          <h3
            class="postTitle"
          >
            Design tips for designers that cover everything you need
          </h3>
        </a>
        <span
          class="postText"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
        </span>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="social-pic"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <span
          class="postDateInfo variable"
        >
          Mocked authorTag
           
          <span>
            Jenny Wilson
          </span>
           
          Aug 23, 2024
        </span>
        <a
          href="/posts/3"
        >
          <h3
            class="postTitle"
          >
            How to build rapport with your web design clients
          </h3>
        </a>
        <span
          class="postText"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
        </span>
      </article>
      <article
        class="postContainer"
      >
        <div
          class="postImageWrapper"
        >
          <img
            alt="social-pic"
            class="postImage"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
        <span
          class="postDateInfo variable"
        >
          Mocked authorTag
           
          <span>
            Leslie Alexander
          </span>
           
          Aug 23, 2024
        </span>
        <a
          href="/posts/4"
        >
          <h3
            class="postTitle"
          >
            Logo design trends to avoid in 2022
          </h3>
        </a>
        <span
          class="postText"
        >
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident.
        </span>
      </article>
    </div>
  </section>
</div>
`;
